partiql-lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt

```diff
@@ -227,14 +236,31 @@ internal class CompilerPipelineImpl(
     private val ion: IonSystem,
     private val parser: Parser,
     override val compileOptions: CompileOptions,
+    override val functionList: List<ExprFunction>,
     override val functions: Map<String, ExprFunction>,
```

Comment:
We really don't need to change the field type. It's unfortunate that it is a map, but there could be some subtle behavior issues. Do we guarantee that a function's key in the map is the same as its name in the signature. Sure we do when we add via the builder, but that's not the case if someone provides the function themselves.

Could you please confirm the behavior of a function whose signature name differs than the one given in the map?